custom_components/jablotron_cloud/alarm_control_panel.py

```diff
@@ -1,208 +1,280 @@
+
+        # Initialize alarm control panel
+        super().__init__(coordinator)
 
     @property
     def code_format(self) -> CodeFormat | None:
         """Disable code for sections that don't require it."""
-        return CodeFormat.NUMBER if self.code_arm_required else None
+
+        return CodeFormat.NUMBER if self._authorization_required else None
 
     @property
     def code_arm_required(self) -> bool:
-        """Whether the code is required for arm actions."""
-        return self._need_authorization
+        """Whether code is required for arm actions."""
+
+        return self._authorization_required
 
     @property
     def supported_features(self) -> AlarmControlPanelEntityFeature:
-        """Return the list of supported features."""
-        if self._can_partial_arm:
-            return (
-                AlarmControlPanelEntityFeature.ARM_AWAY
-                | AlarmControlPanelEntityFeature.ARM_HOME
-            )
+        """Return list of supported features."""
 
-        return AlarmControlPanelEntityFeature.ARM_AWAY
+        supported_features = AlarmControlPanelEntityFeature.ARM_AWAY
+        if self._supports_partial_arm:
+            supported_features |= AlarmControlPanelEntityFeature.ARM_HOME
+
+        return supported_features
 
     @property
     def device_info(self) -> DeviceInfo:
-        """Return the device info."""
+        """Return information about device."""
+
         return DeviceInfo(
-            identifiers={
-                # Serial numbers are unique identifiers within a specific domain
-                (DOMAIN, str(self._service_id))
-            },
-            name=self.coordinator.data[self._service_id]["service"]["name"],
+            identifiers={(DOMAIN, str(self._service_id))},
+            name=self._service_name,
             manufacturer="Jablotron",
-            model=self.coordinator.data[self._service_id]["service"][SERVICE_TYPE],
+            model=self._service_type,
+            sw_version=self._service_firmware
         )
 
     def alarm_disarm(self, code: str | None = None) -> None:
-        """Disarm section."""
-        self._setup_pin(code)
+        """Send disarm request."""
+
+        try:
+            # Send disarm request to section
+            code = self.code_or_default_code(code)
+            bridge = self._client.get_bridge()
+            disarm_successful = bridge.control_section(
+                service_id=self._service_id,
+                service_type=self._service_type,
+                component_id=self._section_id,
+                state="DISARM",
+                pin_code=code
+            )
 
-        self.coordinator.bridge.control_component(
-            self._service_id, self._component_id, Actions.DISARM, self._service_type
-        )
-        self._attr_alarm_state = AlarmControlPanelState.DISARMING
-        self.schedule_update_ha_state()
+            # Set state to disarming if disarm action was successful
+            if disarm_successful:
```

Comment:
What will HA do in case this is False? Do we need to give some error? Same for all the alarm actions.